pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/actions/RestartDeclarativePipelineActionTest.java

```diff
@@ -0,0 +1,709 @@
+import org.junit.Test;
+import org.jvnet.hudson.test.Issue;
+
+import java.util.List;
+
+import static org.jenkinsci.plugins.pipeline.modeldefinition.BasicModelDefTest.stageStatusPredicate;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+public class RestartDeclarativePipelineActionTest extends AbstractModelDefTest {
+
+    @Issue("JENKINS-45455")
+    @Test
+    public void actionPresentOnDeclarative() throws Exception {
+        WorkflowRun r = expect("simplePipeline").go();
+
+        assertNotNull(r.getAction(RestartDeclarativePipelineAction.class));
+    }
+
+    @Issue("JENKINS-45455")
+    @Test
+    public void restartDisabled() throws Exception {
+        SecurityRealm originalRealm = j.jenkins.getSecurityRealm();
+        AuthorizationStrategy originalStrategy = j.jenkins.getAuthorizationStrategy();
+
+        try {
+            j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
+            GlobalMatrixAuthorizationStrategy gmas = new GlobalMatrixAuthorizationStrategy();
+            // Set up an administrator, and three developer users with varying levels of access.
+            gmas.add(Jenkins.ADMINISTER, "admin");
+            gmas.add(Jenkins.READ, "dev1");
+            gmas.add(Item.BUILD, "dev1"); // implies REPLAY
+            gmas.add(Jenkins.READ, "dev2");
+            j.jenkins.setAuthorizationStrategy(gmas);
+            WorkflowJob p = j.jenkins.createProject(WorkflowJob.class, "restartDisabled");
+            p.setDefinition(new CpsFlowDefinition(
+                    "pipeline {\n" +
+                            "  agent any\n" +
+                            "  stages {\n" +
+                            "    stage('before') {\n" +
+                            "      steps {\n" +
+                            "        semaphore 'wait'\n" +
+                            "      }\n" +
+                            "    }\n" +
+                            "  }\n" +
+                            "}\n", true));
+
+            WorkflowRun b = p.scheduleBuild2(0).waitForStart();
+            SemaphoreStep.waitForStart("wait/1", b);
+
+            RestartDeclarativePipelineAction action = b.getAction(RestartDeclarativePipelineAction.class);
+            assertNotNull(action);
+
+            assertFalse(action.isRestartEnabled());
+            SemaphoreStep.success("wait/1", null);
+            j.assertBuildStatusSuccess(j.waitForCompletion(b));
+
+            assertTrue(action.isRestartEnabled());
+            assertTrue(canRestart(b, "admin"));
+            assertTrue(canRestart(b, "dev1"));
+            assertFalse(canRestart(b, "dev2"));
+
+            p.setDisabled(true);
+            assertFalse(canRestart(b, "admin1"));
+            assertFalse(canRestart(b, "dev1"));
+            assertFalse(canRestart(b, "dev2"));
+        } finally {
```

Comment:
Are you using a `ClassRule` or something? With normal `@Rule JenkinsRule` there is no need to reset anything.

Reply:
Yup, I'm using `@ClassRule`.